mac/KeymanEngine4Mac/KeymanEngine4Mac/KME/OnScreenKeyboard/KeyView.m

```diff
@@ -141,6 +144,26 @@ - (void)setKeyCode:(NSUInteger)keyCode {
         _isSpecialKey = YES;
     else
         _isSpecialKey = NO;
+    
+    if (!_isSpecialKey && !_isModifierKey)
+        _isCharacterKey = YES;
+    else
+        _isCharacterKey = NO;
```

Comment:
For new code, use the [style guide](https://github.com/keymanapp/keyman/wiki/Keyman-Code-Style-Guide#code-format) for blocks (although spacing for new code should match file spacing):

```suggestion
    if (!_isSpecialKey && !_isModifierKey) {
        _isCharacterKey = YES;
    } else {
        _isCharacterKey = NO;
    }
```

(If XCode wants `else {` on a new line, we can live with that :wink:)